src/cogent3/core/new_alignment.py

```diff
@@ -3262,6 +3265,8 @@ def map(self) -> IndelMap:
                 numpy.array([], dtype=int),
             )
 
+        # refactor: design
+        # should the returned map be sliced with the slice_record?
```

Comment:
my reaction to this comment is yes. But we need to revisit where the responsibilities lie for doing slicing and make sure we're consistent across the API for all the classes. I would think that the AlignedDataView handles slicing of all the variants of a sequence (str, bytes, array by gapped, ungapped) and thus for the gapped arrays must also handle slicing of the indel map too. I suggest looking at every method that delivers an attribute relied on by Aligned. Each of those should employ the same strategy. If not, let's discuss!